src/rgw/rgw_json_enc.cc

```diff
@@ -1494,3 +1494,70 @@ void RGWOrphanSearchState::dump(Formatter *f) const
   encode_json("stage", stage, f);
   f->close_section();
 }
+
+void RGWObjTags::dump(Formatter *f) const
+{
+  for (const auto& tag: tag_map){
+    f->open_object_section("tag_map");
+    f->dump_string("Key", tag.first);
+    f->dump_string("Value", tag.second);
+    f->close_section();
+  }
+}
+
+void lc_op::dump(Formatter *f) const
+{
+  encode_json("status", status, f);
+  encode_json("dm_expiration", dm_expiration, f);
+  encode_json("expiration", expiration, f);
+  encode_json("noncur_expiration", noncur_expiration, f);
+  encode_json("mp_expiration", mp_expiration, f);
+  if (expiration_date != boost::none) {
+    utime_t ut(*expiration_date);
+    encode_json("expiration_date", ut, f);
+  }
+  if (obj_tags != boost::none) {
+    encode_json("obj_tags", *obj_tags, f);
+  }
+}
+
+void LCFilter::dump(Formatter *f) const
+{
+  encode_json("prefix", prefix, f);
+  encode_json("obj_tags", obj_tags, f);
+}
+
+void LCExpiration::dump(Formatter *f) const
+{
+  encode_json("days", days, f);
+  encode_json("date", date, f);
+}
+
+void LCRule::dump(Formatter *f) const
+{
+  encode_json("id", id, f);
+  encode_json("prefix", prefix, f);
+  encode_json("status", status, f);
+  encode_json("expiration", expiration, f);
+  encode_json("noncur_expiration", noncur_expiration, f);
+  encode_json("mp_expiration", mp_expiration, f);
+  encode_json("filter", filter, f);
+  encode_json("dm_expiration", dm_expiration, f);
+}
+
+void RGWLifecycleConfiguration::dump(Formatter *f) const
+{
+  map<string, lc_op>::const_iterator iter = prefix_map.begin();
+  f->open_object_section("prefix_map");
+  for (; iter != prefix_map.end(); ++iter) {
+    f->dump_object(iter->first.c_str(), iter->second);
+  }
+  f->close_section();
+
+  multimap<string, LCRule>::const_iterator multi_iter = rule_map.begin();
+  f->open_object_section("rule_map");
```

Comment:
here using an object isn't a good idea since it's a multimap<>, so there might be duplicate rule names?

Reply:
@liewegas Thank you,  Please check again. The last change is the map usging an object , and the multimap using an array.